mailbox/backup/src/main/java/org/apache/james/mailbox/backup/MailboxBackup.java

```diff
@@ -0,0 +1,36 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup;
+
+import java.io.IOException;
+
+import org.apache.james.core.User;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.BlobId;
+import org.reactivestreams.Publisher;
+
+public interface MailboxBackup {
+
+    /**
+     * @param user the user account to export
+     * @return a Publisher for the BlobId of the zip archive in the blob store.
+     */
+    Publisher<BlobId> backupAccount(User user) throws IOException, MailboxException;
```

Comment:
I wonder if a `Task` would not make more sense here.

WDYT @mbaechler ?

Reply:
Both describe an async process, Task is probably more specific.
The question could be: will any call of this method be made outside of a Task context?
If not, we should return a Task.
WDYT?

Reply:
IMO it is some pretty heavy information.

With a task we can:

 - Monitor progress
 - Cancel it

Sounds like desirable properties....

Reply:
Publisher can be cancelled. Progress monitoring is a good argument.

Reply:
We want something like a 
Flux<Optional<BlobId>,  TaskExecutionDetails>
with on each step of the process a value is emitted with the execution details updated.
And the last step come with the result.
No ?
The signature exposed could then be a Publisher<Optional<BlobId>,  TaskExecutionDetails> ? 
with the java doc explaining how it works ?

Reply:
We should create another issue for this no ?

Reply:
Ok